app/src/main/java/org/exoplatform/fragment/PlatformWebViewFragment.java

```diff
@@ -392,11 +392,18 @@ public void onResponse(Call call, Response response) throws IOException {
 
     @Override
     public boolean shouldOverrideUrlLoading(WebView view, String url) {
+      // For external and short links, broadcast logout event if done
       if (url.contains(LOGOUT_PATH)) {
         mListener.onUserJustBeforeSignedOut();
       }
-      mListener.onExternalContentRequested(url);
-      return true;
+      if (url != null && url.contains(mServer.getShortUrl())) {
```

Comment:
Switch https://developer.android.com/reference/android/webkit/WebViewClient#shouldOverrideUrlLoading(android.webkit.WebView,%20android.webkit.WebResourceRequest)

> boolean | true to cancel the current load, otherwise return false.

So if it returns false we should use external link, isn't it ?
If yes, the code has to be:

```java
      if (url != null && url.contains(mServer.getShortUrl()) && super.shouldOverrideUrlLoading(view, url)) { // url is on the server's domain, keep loading normally
        return true;
      } else { // url is on an external domain, load in a different fragment
        mListener.onExternalContentRequested(url);
        return true;
      }

```

Reply:
Yes, if it returns false we should use external link.I added this change but it doesn't works as expected.
it always return false and use external link, as a result the download button doesn't works.
IMO we should keep these changes. WDYT?

Reply:
Function has been updated